picker_wrapper.go

```diff
@@ -245,20 +247,20 @@ type stickyStore struct {
 	// curMDKey is check before every get/put to avoid races. The operation will
 	// abort immediately when the given mdKey is different from the curMDKey.
 	curMDKey string
-	store    map[string]*stickyStoreEntry
+	store    *linkedMap
 }
 
 func newStickyStore() *stickyStore {
 	return &stickyStore{
-		store: make(map[string]*stickyStoreEntry),
+		store: newLinkedMap(),
 	}
 }
 
 // reset clears the map in stickyStore, and set the currentMDKey to newMDKey.
 func (ss *stickyStore) reset(newMDKey string) {
 	ss.mu.Lock()
 	ss.curMDKey = newMDKey
-	ss.store = make(map[string]*stickyStoreEntry)
+	ss.store = newLinkedMap()
```

Comment:
`linkedMap` itself has a `clear` method. Should we just call that instead?

Reply:
Done!